### PR TITLE
Update readme.md and build.hpp

### PR DESCRIPTION
**Pull request by qureshinomaan**
Updating tutorial as mentioned in 
https://github.com/mlpack/mlpack.org/pull/23

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -123,10 +123,11 @@ If you are compiling Armadillo by hand, ensure that LAPACK and BLAS are enabled.
 
 ### 4. Building mlpack from source
 
-This section discusses how to build mlpack from source.  However, mlpack is in
-the repositories of many Linux distributions, so it may be easier to use the
-package manager for your system.  For example, on Ubuntu, you can install mlpack
-with the following command:
+This document discusses how to build mlpack from source. These build directions 
+will work for any Linux-like shell environment (for example Ubuntu, macOS,
+freeBSD etc). However, mlpack is in the repositories of many Linux distributions
```
Comment (zoq) [inline README.md]: Minor issue it's FreeBSD.

Comment (qureshinomaan) [reply README.md]: Done!

File: doc/guide/build.hpp [hunk 2]
```diff
@@ -227,7 +228,8 @@ and submit an issue and the mlpack developers will quickly help you figure it
 
 https://github.com/mlpack/mlpack
 
-Alternately, mlpack help can be found in IRC at \#mlpack on irc.freenode.net.
+Alternately, mlpack help can be found in IRC at \#mlpack on irc.
```
Comment (zoq) [inline doc/guide/build.hpp]: Isn't it `chat.freenode.net`?

**Comment by qureshinomaan** [comment]
@zoq, @rcurtin Please review this pull request. Does anything else need to be changed?

File: doc/guide/build.hpp [hunk 3]
```diff
@@ -228,8 +228,7 @@ and submit an issue and the mlpack developers will quickly help you figure it
 
 https://github.com/mlpack/mlpack
 
-Alternately, mlpack help can be found in IRC at \#mlpack on irc.
-node.net.
+Alternately, mlpack help can be found in IRC at \#mlpack on irc.freenode.net.
```
Comment (zoq) [inline doc/guide/build.hpp]: Pretty sure `irc.freenode.net` doesn't work anymore, but `chat.freenode.net` does which is also mentioned on the website.

Comment (qureshinomaan) [reply doc/guide/build.hpp]: Done!

**Comment by zoq** [comment]
Thanks again for the contribution.

**Pull request merged**